quic-client/src/lib.rs

```diff
@@ -140,9 +142,14 @@ impl QuicConfig {
     pub fn update_client_certificate(
         &mut self,
         keypair: &Keypair,
-        ipaddr: IpAddr,
+        ipaddr: Option<IpAddr>,
```

Comment:
The `Option<IpAddr>` is ambiguous where it is fine to have `None` and where it is not.

Instead of overloading this method, can we keep `ipaddr: IpAddr` here, and instead add a separate `update_keypair` method which does not take `ipaddr` argument?

Also doing so, the `update_keypair` method will only change `self.client_certficate`.
If `self.client_certificate` is wrapped in `RwLock` then the method receiver will be an immutable `&self`.
That would remove the need to wrap the `ConnectionManger` in a `RwLock`.